### ISSUE TITLE
Add cluster field to whitelists/blacklists

### ISSUE THREAD

**Issue by DirtyCajunRice** [argoproj/argo-cd#4128]
# Summary

In projects that are allowed to deploy to > 1 cluster, whitelists/blacklists are scoped to all clusters

# Motivation

Allow a project to deploy app of apps, in which they need access not only to their own cluster, but to the argocd hosted cluster. We would like to allow them to deploy a larger whitelist to their own cluster, but only an Application CRD to the argocd cluster.

# Proposal

Add the ability to define cluster(s) in the whitelist/blacklist rule. Defaulting to * or empty will ensure backwards compatibility.

**Comment by mleklund** [comment]
Yes! I would love to be able to add ArgoCD Applications from any repo, but not touch the rest of the items in the argocd namespace.  Thus allowing each argo app to manage itself.  Another option would be to allow Argo to watch for application CRDs in any namespace.
